grid/grid_network.py

```diff
@@ -55,19 +114,98 @@ def serve_model(self, model, model_id):
             host_worker.serve_model(model, model_id=model_id)
             host_worker.disconnect()
 
-    def run_inference(self, model_id, dataset):
+    def run_inference(self, model_id, dataset, encrypted=False, copy=True):
```

Comment:
Same about having 2 functions: maybe have run_inference and run_encrypted_inference